### PR TITLE
Deployer module code analysis fix for handlers

### PR DESCRIPTION
**Pull request by tushar-tushar**

### CODE REVIEW

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/broker/K8sResourceBroker.java [hunk 1]
```diff
@@ -61,29 +61,29 @@ public void withNamespace(String namespace) {
         }
         V1ObjectMeta objectMeta = kubernetesResource.getV1ObjectMeta();
         T obj = (T) kubernetesResource.getResource();
-        String namespace = kubernetesResource.getV1ObjectMeta().getNamespace();
+        String resourceNamespace = kubernetesResource.getV1ObjectMeta().getNamespace();
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/broker/K8sResourceBroker.java:64]: Why this change ?

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/config/DeployerEnvConfig.java [hunk 2]
```diff
@@ -22,6 +22,8 @@
 public class DeployerEnvConfig {
 
     private static Logger logger = LoggerFactory.getLogger(DeployerEnvConfig.class);
+    
+    private DeployerEnvConfig() {}
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/config/DeployerEnvConfig.java:26]: Instead make the class itself a static class

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/builder/NamespaceBuilder.java [hunk 3]
```diff
@@ -24,6 +24,8 @@
  *
  */
 public class NamespaceBuilder {
+    
+    private NamespaceBuilder() {}
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/builder/NamespaceBuilder.java:28]: Make the class static instead

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/factory/PodParentFactory.java [hunk 4]
```diff
@@ -29,13 +29,15 @@
  *
  */
 public class PodParentFactory {
+    
+    private PodParentFactory() {}
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/factory/PodParentFactory.java:33]: static class

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/manager/AnnotationsUpdateManager.java [hunk 5]
```diff
@@ -22,6 +22,8 @@
 import java.util.HashMap;
 
 public class AnnotationsUpdateManager {
+    
+    private AnnotationsUpdateManager() {}
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/manager/AnnotationsUpdateManager.java:26]: static class

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/model/PodStatusCode.java [hunk 6]
```diff
@@ -19,8 +19,10 @@
 import java.util.Map;
 
 public class PodStatusCode {
+    
+    private PodStatusCode() {}
 
-	public static Map<Integer, Signals> statusCodeVsMessage = new HashMap<>();
+	private static final Map<Integer, Signals> statusCodeVsMessage = new HashMap<>();
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/model/PodStatusCode.java:25]: Not necessary to be outside ! This can sit inside the Signals enum itself.
The method should be indicate of podstatus code.

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/DeploymentStatusUtil.java [hunk 7]
```diff
@@ -31,6 +31,8 @@
  *	Utility for Service Deployment status
  */
 public class DeploymentStatusUtil {
+    
+    private DeploymentStatusUtil() {}
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/DeploymentStatusUtil.java:35]: static class

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sDeployerUtil.java [hunk 8]
```diff
@@ -43,6 +43,8 @@
 public class K8sDeployerUtil {
 
     private static final Logger logger = LoggerFactory.getLogger(K8sDeployerUtil.class);
+    
+    private K8sDeployerUtil() {}
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sDeployerUtil.java:47]: static class

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sPodUtil.java [hunk 9]
```diff
@@ -38,6 +38,7 @@
  */
 public class K8sPodUtil {
 
+    private K8sPodUtil() {}
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sPodUtil.java:41]: static class

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sReplicaUtil.java [hunk 10]
```diff
@@ -23,16 +24,16 @@
 
 public class K8sReplicaUtil {
 
-
+    private K8sReplicaUtil() {}
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sReplicaUtil.java:27]: static class

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sResourcePatchUtil.java [hunk 11]
```diff
@@ -33,6 +34,8 @@
  */
 public class K8sResourcePatchUtil {
 
+    private K8sResourcePatchUtil() {}
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/K8sResourcePatchUtil.java:37]: Static class

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/KubernetesVolumeUtil.java [hunk 12]
```diff
@@ -40,19 +41,21 @@
  *
  */
 public class KubernetesVolumeUtil {
+    
+    private KubernetesVolumeUtil() {}
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/util/KubernetesVolumeUtil.java:45]: static

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/broker/K8sResourceBroker.java (hunk 1 above)
Comment (tushar-tushar) [reply deployer/deployer-services/src/main/java/io/hyscale/deployer/services/broker/K8sResourceBroker.java:64]: In class we already have a namespace field, which as per sonar analysis is a major code smell.

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/model/ServiceAddress.java [hunk 13]
```diff
@@ -56,7 +56,7 @@ public void setPorts(List<Integer> ports) {
     @Override
     public String toString() {
 	if (StringUtils.isBlank(serviceIP)) {
-	    return null;
+	    return "";
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/model/ServiceAddress.java:59]: use StringUtils.EMPTY

File: deployer/deployer-services/src/main/java/io/hyscale/deployer/services/model/PodStatusCode.java [hunk 14]
```diff
@@ -19,20 +19,19 @@
...
-
-	private static final Map<Integer, Signals> statusCodeVsMessage = new HashMap<>();
-
-	static {
-		for (Signals each : Signals.values()) {
-			statusCodeVsMessage.put(each.getCode(), each);
-		}
-	}
-
+
```
Comment (SameeraGrandhi) [inline deployer/deployer-services/src/main/java/io/hyscale/deployer/services/model/PodStatusCode.java:23]: So what I meant is to remove this class and just rely on the Signals enum. The method which fetches code should be indicative.
